### PR TITLE
Feature/project has to have valid name

### PR DESCRIPTION
**Pull request by DavidDamke**
grafik
grafik

### CODE REVIEW

**Review by Death111 — CHANGES_REQUESTED** [review]

File: src/main/java/de/doubleslash/keeptime/view/ManageProjectController.java [hunk 1]
```diff
@@ -78,6 +83,16 @@ public void initializeWith(final Project project) {
       isWorkCheckBox.setSelected(project.isWork());
       sortIndexSpinner.getValueFactory().setValue(project.getIndex());
    }
+   public void validateName(){
+      if(nameTextField.getText().isBlank()){
```
[the remark this answers is not in the document]
Comment (DavidDamke) [reply src/main/java/de/doubleslash/keeptime/view/ManageProjectController.java:99]: Implemented the property bindings, but not sure if I did it correctly

File: src/main/java/de/doubleslash/keeptime/view/ManageProjectController.java [hunk 2]
```diff
@@ -57,6 +56,10 @@ public class ManageProjectController {
    @FXML
    private Spinner<Integer> sortIndexSpinner;
 
+   @FXML
+   private Label validateTextAlert;
+
+   public static BooleanProperty projectNameIsValid = new SimpleBooleanProperty(false);
```
[the remark this answers is not in the document]
Comment (DavidDamke) [reply src/main/java/de/doubleslash/keeptime/view/ManageProjectController.java]: added a getter for the Property

File: src/main/java/de/doubleslash/keeptime/view/ManageProjectController.java [hunk 3]
```diff
@@ -68,6 +71,8 @@ private void initialize() {
       sortIndexSpinner
             .setValueFactory(new IntegerSpinnerValueFactory(0, availableProjectAmount, availableProjectAmount));
       sortIndexSpinner.getValueFactory().setValue(model.getAvailableProjects().size());
+      projectNameIsValid.set(false);
+      nameTextField.setOnKeyTyped(event -> validateName());
```
[the remark this answers is not in the document]
Comment (DavidDamke) [reply src/main/java/de/doubleslash/keeptime/view/ManageProjectController.java]: added property binding

File: src/main/java/de/doubleslash/keeptime/view/ManageProjectController.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Death111) [reply src/main/java/de/doubleslash/keeptime/view/ManageProjectController.java:91]: let's have a look together

File: src/main/resources/layouts/manage-project.fxml [hunk 4]
```diff
@@ -25,23 +25,25 @@
...
+      <CheckBox fx:id="isWorkCheckBox" mnemonicParsing="false" selected="true" GridPane.columnIndex="1" GridPane.rowIndex="5" />
+      <Spinner fx:id="sortIndexSpinner" GridPane.columnIndex="1" GridPane.rowIndex="6" />
+      <TextArea fx:id="descriptionTextArea" prefHeight="120.0" prefWidth="300.0" GridPane.columnIndex="1" GridPane.rowIndex="2" />
+      <VBox alignment="CENTER_LEFT" GridPane.rowIndex="2">
          <children>
             <Label text="Description:" />
             <Label disable="true" prefWidth="100.0" text="(will be matched at search)" wrapText="true" />
          </children>
       </VBox>
+      <Label fx:id="validateTextAlert" prefHeight="17.0" prefWidth="201.0" text="Please enter a project name!" GridPane.columnIndex="1" GridPane.rowIndex="1" />
```
Comment (Death111) [inline src/main/resources/layouts/manage-project.fxml:46]: let's remove the "!" from the hint :) Sounds more friendly then - especially as the hint appears directly when opening the dialog

File: src/main/java/de/doubleslash/keeptime/view/ViewController.java [hunk 5]
```diff
@@ -505,6 +504,8 @@ private GridPane setUpAddNewProjectGridPane(final Dialog<Project> dialog) {
 
       final ManageProjectController manageProjectController = loader.getController();
 
+      dialog.getDialogPane().lookupButton(ButtonType.OK).disableProperty().bind(manageProjectController.formValidProperty().not());
```
Comment (Death111) [inline src/main/java/de/doubleslash/keeptime/view/ViewController.java:507]: when editing a project, the "hint" to set a name also shows. but the button is not disabled.
should disable it also while editing the project. maybe we can reuse the code somehow?

**Review by Death111 — CHANGES_REQUESTED** [review]

File: src/main/java/de/doubleslash/keeptime/view/ViewController.java (hunk 5 above)
Comment (DavidDamke) [reply src/main/java/de/doubleslash/keeptime/view/ViewController.java:507]: Button is now disabled

File: src/main/resources/layouts/manage-project.fxml (hunk 4 above)
Comment (DavidDamke) [reply src/main/resources/layouts/manage-project.fxml:46]: Removed it :)

**Review by Death111 — APPROVED** [review]

**Pull request merged**